lib/core/output_sink/es_sink.rb

```diff
@@ -15,7 +15,7 @@ module Core::OutputSink
   class EsSink < Core::OutputSink::BaseSink
     def initialize(index_name, flush_threshold = 50)
       super()
-      @client = Utility::EsClient.new
+      @client = Utility::EsClient.new(App::Config[:elasticsearch])
```

Comment:
I'm thinking, what is the difference if we change it to `Elasticsearch::Client.new(App::Config[:elasticsearch])`?

We initially have `Utility::EsClient` because we only need one instance of ES client. With multi-connectors support, I think we are going to create a new instance of ES client for each sync job.

Reply:
There's one feature that we've added on top of `bulk`: 

```
def bulk(arguments = {})
      raise_if_necessary(super(arguments))
    end

    private

    def raise_if_necessary(response)
      if response['errors']
        first_error = response['items'][0]
        raise IndexingFailedError.new("Failed to index documents into Elasticsearch.\nFirst error in response is: #{first_error.to_json}")
      end
      response
    end
```

Removing the client fully will make us do this logic somewhere else, which is not a problem, but something to consider during refactoring.